### PR TITLE
Filter execution end-points (management and internal)

### PR DESCRIPTION
**Pull request by tranchitella**
Backport the filter execution end-points, both management and internal,
from the enterprise version of inventory.

Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: api/http/api_inventory.go [hunk 1]
```diff
@@ -588,6 +598,69 @@ func (i *inventoryHandlers) CreateTenantHandler(w rest.ResponseWriter, r *rest.R
...
+	if err != nil {
+		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		return
+	}
+
+	// query database
+	devs, totalCount, err := i.inventory.SearchDevices(ctx, *searchParams)
+	if err != nil {
+		if strings.Contains(err.Error(), "BadValue") {
+			u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
```
Comment (merlin-northern) [inline api/http/api_inventory.go:617]: adding this unit test scenario will make codecov happy.

File: api/http/api_inventory.go [hunk 2]
```diff
@@ -588,6 +598,69 @@ func (i *inventoryHandlers) CreateTenantHandler(w rest.ResponseWriter, r *rest.R
...
```
Comment (merlin-northern) [inline api/http/api_inventory.go:652]: adding this unit test scenario will make codecov happy.

File: api/http/api_inventory.go [hunk 3]
```diff
@@ -643,3 +716,24 @@ func (i *inventoryHandlers) InternalDevicesStatusHandler(w rest.ResponseWriter,
 
 	w.WriteHeader(http.StatusOK)
 }
+
+func parseSearchParams(r *rest.Request) (*model.SearchParams, error) {
+	var searchParams model.SearchParams
+
+	if err := r.DecodeJsonPayload(&searchParams); err != nil {
+		return nil, errors.Wrap(err, "failed to decode request body")
```
Comment (merlin-northern) [inline api/http/api_inventory.go:724]: adding this unit test scenario will make codecov happy.

File: api/http/api_inventory.go (hunk 1 above)
Comment (tranchitella) [reply api/http/api_inventory.go:617]: done

File: api/http/api_inventory.go (hunk 2 above)
Comment (tranchitella) [reply api/http/api_inventory.go:652]: done

File: api/http/api_inventory.go (hunk 3 above)
Comment (tranchitella) [reply api/http/api_inventory.go:724]: done

**Comment by tranchitella** [comment]
Stopped the pipeline, the force-push only contains updated unit tests.

**Pull request merged**